### PR TITLE
Decrease cyclomatic complexity

### PR DESCRIPTION
**Pull request by afbjorklund**
Mostly by using the "extract method" pattern.

Here was the report (from gocyclo), before:

```
44 tunnel TestTunnel pkg/minikube/tunnel/tunnel_test.go:35:1
34 kubeconfig configEquals pkg/util/kubeconfig/config_test.go:420:1
25 integration testMounting test/integration/mount_test.go:36:1
21 util setElement pkg/util/config.go:50:1
18 integration testTunnel test/integration/tunnel_test.go:37:1
17 tunnel TestTunnelManagerCleanup pkg/minikube/tunnel/tunnel_manager_test.go:174:1
16 cluster StartHost pkg/minikube/cluster/cluster.go:64:1
16 kubeadm (*KubeadmBootstrapper).UpdateCluster pkg/minikube/bootstrapper/kubeadm/kubeadm.go:374:1
16 provision configureAuth pkg/provision/buildroot.go:218:1
16 tunnel setupRoute pkg/minikube/tunnel/tunnel.go:135:1
```

### CODE REVIEW

**Comment by afbjorklund** [comment]
Hmm, hard to build the integration tests locally apparently...

File: pkg/util/kubeconfig/config_test.go [hunk 1]
```diff
@@ -434,6 +434,24 @@ func configEquals(a, b *api.Config) bool {
 	}
 
 	// clusters
+	if !clustersEquals(a, b) {
```
[the remark this answers is not in the document]
Comment (afbjorklund) [reply pkg/util/kubeconfig/config_test.go]: That might be possible, I just copied the code without improving.

**Comment by afbjorklund** [comment]
There's also a new warning that needs fixing too.
```
	cmd/minikube/cmd/config/util.go
		Line 105: warning: cyclomatic complexity 16 of function EnableOrDisableAddon() is high (> 15) (gocyclo)
		Line 105: warning: cyclomatic complexity 16 of function EnableOrDisableAddon() is high (> 15) (gocyclo)
		Line 105: warning: cyclomatic complexity 16 of function EnableOrDisableAddon() is high (> 15) (gocyclo)
```

File: pkg/util/kubeconfig/config_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (afbjorklund) [reply pkg/util/kubeconfig/config_test.go]: I tried replacing it, but then the test started failing. Apparently not all fields are used ?

`invalid value "foo" for flag -e: Invalid value for ExtraOption flag. Value must contain at least one period: foo` (_and so on_)

```
--- FAIL: TestNewConfig (0.00s)
    kubeconfig_test.go:332: configs did not match

```

**Comment by tstromberg** [comment]
Looks good. Please merge once rebased.

**Pull request merged**